src/utils/yamlUtils.ts

```diff
@@ -0,0 +1,66 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.txt in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+
+/* eslint-disable @typescript-eslint/no-unsafe-call, @typescript-eslint/no-explicit-any, @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-member-access, no-prototype-builtins */
+
+import { readFile } from "fs-extra";
+import { basename } from "path";
+import { ext } from "vscode-azureappservice/out/src/extensionVariables";
+import { parse } from "yaml";
+import { BuildConfig } from "../tree/localProject/ConfigGroupTreeItem";
+import { localize } from "./localize";
+
+export async function parseYamlFile(yamlFilePath: string): Promise<Map<BuildConfig, string> | undefined> {
+    const contents: string = (await readFile(yamlFilePath)).toString();
+
+    if (isSwaYaml(contents)) {
+        const parsedYaml: any = await parse(contents);
+
+        for (const job of <any[]>Object.values(parsedYaml.jobs)) {
+            for (const step of <any[]>Object.values(job['steps'])) {
+                if (isBuildDeployStep(step)) {
+                    return getBuildConfigs(step, basename(yamlFilePath));
+                }
+            }
+        }
+    }
+
+    return undefined;
+}
+
+function isSwaYaml(yamlFileContents: string): boolean {
+    return /Azure\/static-web-apps-deploy/.test(yamlFileContents);
+}
+
+function isBuildDeployStep(step: any): boolean {
+    return step.hasOwnProperty('id') && step['id'] === 'builddeploy';
+}
+
+function getBuildConfigs(step: any, yamlFileName: string): Map<BuildConfig, string> | undefined {
```

Comment:
I think we can break this down a bit more.  If we make the BuildConfig a parameter, you can replace some of the repeat code such as

```
!step['with'].hasOwnProperty('api_location') || !step['with'].hasOwnProperty('app_location'))
 buildConfigs.set('api_location', getBuildConfig(step, 'api_location'))
    buildConfigs.set('app_location', getBuildConfig(step, 'app_location'))
void ext.ui.showWarningMessage(localize('mustIncludeLocations', `"{0}" must include "api_location" and "app_location". See the [workflow file guide](https://aka.ms/AAbrcox).`, yamlFileName));
```

And then just have it return the string values and set the map when you call `getBuildConfigs`.  To test for `output/app_artifact`, you can just check if the map has `output` defined (or has 3 settings) before checking for `app_artifact`